pint_xarray/conversion.py

```diff
@@ -121,28 +122,62 @@ def attach_units_variable(variable, units):
     return new_obj
 
 
-def dataset_from_variables(variables, coords, attrs):
-    data_vars = {name: var for name, var in variables.items() if name not in coords}
-    coords = {name: var for name, var in variables.items() if name in coords}
+def dataset_from_variables(variables, coordinate_names, indexes, attrs):
+    data_vars = {
+        name: var for name, var in variables.items() if name not in coordinate_names
+    }
+    coords = {name: var for name, var in variables.items() if name in coordinate_names}
+
+    new_coords = Coordinates._construct_direct(coords, indexes)
```

Comment:
Why do you need to use the private `._construct_direct`? I'm using the public `__init__` just fine

https://github.com/zarr-developers/VirtualiZarr/blob/91ebefe495961b25d50cc1cde29607ddfe291095/virtualizarr/xarray.py#L378

Reply:
that was from before you did the changes to the constructor of `Coordinates`, so it's simply something I forgot to change.